server.py

```diff
@@ -228,3 +231,26 @@ def semesterIDType(reqstr):
 
   # Convert Officer objects back to dictionaries before returning
   return jsonify([x.__dict__ for x in officers])
+
+#I'm keeping these global variables here for now so it's easier to determine that it goes with the events api when the pieces of the api are inevitably split up
+ev = []
+eventsCached = datetime.now()
+@app.route('/api/events')
+def getEvents():
+  calurl = 'https://calendar.google.com/calendar/ical/ca149os3pmnh0dcopr1jn2negg%40group.calendar.google.com/public/basic.ics'
+  dtstrformat = '%Y-%m-%dT%H:%M:%S.%fZ'
+  global ev
+  global eventsCached
+
+  if (ev == []) or (datetime.now() - eventsCached >= timedelta(hours=1)):
+    gcal = Calendar.from_ical(requests.get(calurl).content)
+    ev = []
+    for comp in [x for x in gcal.walk() if x.name=="VEVENT"]:
+      ev.append({'summary': comp.decoded('SUMMARY').decode('UTF-8'), 
+        'timeStart': comp.decoded('DTSTART').strftime(dtstrformat),
+        'timeEnd': comp.decoded('DTEND').strftime(dtstrformat),
+        'location': comp.decoded('LOCATION').decode('UTF-8'),
+        'description': comp.decoded('DESCRIPTION').decode('UTF-8')})
+    eventsCached = datetime.now()
```

Comment:
Can you go through what this is doing?

Reply:
Specifically wrt the ical format stuff.

Reply:
"comp" represents an event. The "decoded" method returns either a binary string or, in the case of start and end dates, a python datetime object. jsonify doesn't work with binary strings so those have to be decoded to raw strings, that's what the "decode('UTF-8')" is for. As for the datetime objects, those have a "strftime" method to format to a string.